include/s3select_oper.h

```diff
@@ -1464,32 +1475,32 @@ class logical_operand : public base_statement
     value a = l->eval();
     if (_oplog == oplog_t::AND)
     {
-      if (a.i64() == false)
-      {
+      if (!a.is_null() && a.i64() == false) {
         res = false ^ negation_result;
         return var_value = res;
       } 
       value b = r->eval();
+      res =  (a.i64() && b.i64()) ^ negation_result ;
```

Comment:
where ((null>4) and (1==0)) is null   ==> return true , that is not correct
should be
right side is false for /AND/ ==> false